### ISSUE TITLE
Disallow requesting `--ir-optimized` output without `--optimize`

### ISSUE THREAD

**Issue by cameel** [ethereum/solidity#13801]
The `--ir-optimized` option on the command-line sounds as if it would give you the optimized IR but that's only true when `--optimize` is also specified. Without it, you get the output that optimizer would get but without any optimizations actually applied. This is very confusing to users.

We should report an error on the CLI when `--ir-optimized` is specified but `--optimize` is not.

We have a similar situation with the `ir` and `irOptimized` outputs in Standard JSON, though I think there it's less of a problem because tooling has this figured out and individual users are not likely to use this directly. The current documentation is not very clear about this behavior though so we should improve it:
```javascript
//   ir - Yul intermediate representation of the code before optimization
//   irOptimized - Intermediate representation after optimization
```

## Backwards Compatibility
This is technically not backwards-compatible, though the combination is pretty useless so I don't think it would affect practical use cases. It would be much more problematic if we decided to do this in Standard JSON too though.

**Comment by cameel** [comment]
I gave it `low impact` because of the assumption that people aren't using the compiler on the CLI very often. I think this does not affect many people overall, but for those affected the impact is definitely higher.

**Comment by aarlt** [comment]
I would say we should not just close this, because it's really an annoying behaviour.

@ekpyron shouldn't we at least leave this issue open?

**Comment by ekpyron** [comment]
It's not like I closed it :-). I just didn't intervene when it was closed via bot, since I'm not sure we need a separate issue for this, but we definitely want this changed eventually, so I'm reopening and relabeling this, s.t. it stays open.
Generally, I'd consider it as part of https://github.com/ethereum/solidity/issues/9583, i.e. the rewrite of the CLI with Standard-JSON backend, although that issue has fallen a bit out of date with our most recent discussions around it (see https://matrix.to/#/#sld-cli-redesign:matrix.org)

So yeah, we can keep this issue open and linked to https://github.com/ethereum/solidity/issues/9583, but I would **not** start to do this separately from doing the bulk transition involved in https://github.com/ethereum/solidity/issues/9583
